columnphysics/icepack_orbital.F90

```diff
@@ -177,11 +177,24 @@ subroutine compute_coszen (tlat,          tlon,        &
 
       real (kind=dbl_kind) :: ydayp1 ! day of year plus one time step
 
+      logical (kind=log_kind), save :: &
+         first_call = .true.   ! first call flag
+
       character(len=*),parameter :: subname='(compute_coszen)'
 
 ! Solar declination for next time step
 
 #ifdef CESMCOUPLED
+      if (icepack_chkoptargflag(first_call)) then
+         if (.not.(present(days_per_year) .and. &
+                   present(nextsw_cday)   .and. &
+                   present(calendar_type))) then
+            call icepack_warnings_add(subname//' error in CESMCOUPLED args')
+            call icepack_warnings_setabort(.true.,__FILE__,__LINE__)
+            return
+         endif
+      endif
+
```

Comment:
OK, so if I understand correctly these flags are not optional for CESM, so we add a check to that effect here?

If so, that makes sense.

Reply:
This might complicate coupling in E3SM, if it's using CESMCOUPLED (I'm not sure).  @njeffery

Reply:
Right, there are very few CESMCOUPLED cpps left in Icepack, we tried to clear them out a while ago and shift more to namelist settings.  Use of CESMCOUPLED in icepack_orbital was discussed of that at the time.  If this causes E3SM a problem, we can refactor it.  The problem is that it will probably require a change in all the models using CESMCOUPLED in terms of how the orbital stuff is handled, so we're trying not to change that until we have to.  But maybe we should do this at some point with consultation with CESM, RASM, etc so it's well coordinated.  New implementation still TBD as well.